File: 0881-boats-to-save-people/0881-boats-to-save-people.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
  people.sort((a, b) => a - b);
  let [left, right, count] = [0, people.length - 1, 0];
  
  while (left <= right) {
    if (people[left] + people[right] <= limit) left++;
    
    right--;
    count++;
  }
  
  return count;
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        lo, hi, ans = 0, len(people) - 1, 0
        while lo <= hi:
            if people[lo] + people[hi] <= limit:
                lo += 1
            hi -= 1
            ans += 1
        return ans
        
                